pkg/controlplane/peer/client.go

```diff
@@ -44,24 +45,60 @@ type RemoteServerAuthorizationResponse struct {
 	AccessToken string
 }
 
-// authorize a request for accessing a peer exported service, yielding an access token.
-func (c *Client) Authorize(req *api.AuthorizationRequest) (*RemoteServerAuthorizationResponse, error) {
-	body, err := json.Marshal(req)
-	if err != nil {
-		return nil, fmt.Errorf("unable to serialize authorization request: %w", err)
+func (c *Client) getResponse(
+	getRespFunc func(client *jsonapi.Client) (*jsonapi.Response, error),
+) (*jsonapi.Response, error) {
+	if len(c.clients) == 1 {
```

Comment:
Why do you need this case and can't do iteration on 1 client?

Reply:
This is the common case, and the purpose is to make it more simple since you don't need go-routines and the multi-threaded syncing in this case.